onnxruntime/core/platform/windows/env.cc

```diff
@@ -124,52 +147,171 @@ class WindowsEnv : public Env {
   }
 
   int GetNumCpuCores() const override {
-    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-    DWORD returnLength = sizeof(buffer);
-    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-      // try GetSystemInfo
-      SYSTEM_INFO sysInfo;
-      GetSystemInfo(&sysInfo);
-      if (sysInfo.dwNumberOfProcessors <= 0) {
-        ORT_THROW("Fatal error: 0 count processors from GetSystemInfo");
+    int num_cores = 0;
+    for (const auto& group_info : group_info_vec_) {
+      num_cores += group_info.ActiveProcessorCount;
+    }
+    return num_cores;
+  }
+
+  size_t GetDefaultThreadpoolSetting(std::vector<uint64_t>& affinities) const override {
+    ORT_ENFORCE(sizeof(KAFFINITY) <= sizeof(uint64_t), "KAFFINITY is bigger than uint64_t, cannot save it to an uint64_t vector");
+    affinities.clear();
+    KAFFINITY group_id = 0;
+    for (const auto& group_info : group_info_vec_) {
+      size_t bit_offset = 0;
+      constexpr size_t bit_limit = sizeof(KAFFINITY) << 3;
+      KAFFINITY active_processor_mask = group_info.ActiveProcessorMask;
+      for (int i = 0; i < (group_info.ActiveProcessorCount >> 1); ++i) {
+        KAFFINITY processor_mask = 0;
+        size_t processor_count = 0;
+        while (processor_count < 2 && bit_offset < bit_limit) {
```

Comment:
>processor_count < 2 [](http://example.com/codeflow?start=15&length=19)

Why are we limiting processor_count to 2?
Also would it be better to name it logical_process_count?
I think the bit_offset would automatically limit the max processor count to 64 in this case.